chapters/VK_KHR_surface/wsi.txt

```diff
@@ -1207,6 +1207,29 @@ include::{generated}/api/enums/VkPresentModeKHR.txt[]
     New requests are appended to the end of the queue, and one request is
     removed from the beginning of the queue and processed during or after
     each vertical blanking period in which the queue is non-empty.
+ifdef::VK_EXT_present_timing[]
+  * ename:VK_PRESENT_MODE_FIFO_LATEST_READY_EXT specifies that the
+    presentation engine waits for the next vertical blanking period to
+    update the current image.
+    Tearing cannot: be observed.
+    An internal queue is used to hold pending presentation requests.
+    New requests are appended to the end of the queue.
+    At each vertical blanking period, the presentation engine dequeues
+    successive-queued images for which their associated wait semaphores have
+    signaled.
+    The last image dequeued is presented.
```

Comment:
Isn't this the same as MAILBOX, but more precisely specified?

Reply:
When does this dequeue operation happen? In the vblank handler itself, or some unspecified time before vblank for next frame (i.e. when the compositor might want to do some work)?

Reply:
As per previous internal discussions, this new present mode will be the subject of a separate extension. The key difference with MAILBOX, I think, is that the dequeue is done at vblank time, instead of when the wait semaphores have been signaled, which is difficult to implement in some situations. I'll make sure these questions are addressed when a proposal comes up.